File: sem3/map/Lab/ToyLanguage/src/Model/Statements/CompStmt.java
package Model.Statements;

import Model.ADT.MyIStack;
import Model.PrgState;

/**
 * Created by Andrei on 20.10.2017.
 */
public class CompStmt implements IStatement {
    private IStatement first;
    private IStatement second;

    public CompStmt(IStatement first, IStatement second) {
        this.first = first;
        this.second = second;
    }

    @Override
    public PrgState execute(PrgState state) {
        MyIStack<IStatement> stack = state.getExeStack();
        stack.push(second);
        stack.push(first);
        return state;
    }

    @Override
    public String toString() {
        return first.toString() + "; " + second.toString();
    }
}

File: sem3/sem 3/~craciunf/c-sharp-toy-language-interpreter/models/CompStmt.cs
/* package models; */
using System;
using System.Collections.Generic;

namespace ToyLanguageInterpreter {
    public class CompStmt : IStmt {
        private IStmt fst, snd;

        public CompStmt(IStmt fst, IStmt snd) {
            this.fst = fst;
            this.snd = snd;
        }

        /* @Override */
        public  String toString() {
            return "(" + this.fst.toString() + ";\n" + this.snd.toString() + ")";
        }

        /* @Override */
        public  PrgState execute(PrgState state) {
            Stack <IStmt> stack = state.getExeStack();

            stack.Push(this.snd);
            stack.Push(this.fst);

            return state;
        }
    }
}
